### ISSUE TITLE
Certain recipes are... backwards?

### ISSUE THREAD

**Issue by FloppyDingo** [MatrexsVigil/harvestcraft#275]
Recipes for certain baking items such as baked sweet potatoes, and popcorn appear to be backwards, wherein the recipe is:

1 popcorn + baking sheet = 1 corn
1 baked sweet potato + baking sheet = 1 sweet potato

**Comment by CubeFromSpace** [comment]
This is because most recipes are shapeless.

**Comment by Syndaryl** [comment]
I'm pretty sure shapelessness has nothing to do with being able to "uncook" popcorn into corn by baking it on the bakeware.

**Comment by CubeFromSpace** [comment]
Then this is a much bigger bug. I'll look at the code

**Comment by CubeFromSpace** [comment]
This may be due to how every recipe is squeezed into this file: https://github.com/MatrexsVigil/harvestcraft/blob/master/src/main/java/com/pam/harvestcraft/item/RecipeRegistry.java

I usually see mods with items all separate files, each including the recipe. This may be a coding typo, or a flaw in Forge/MCP syntax.

**Comment by Peter8Dev** [comment]
Which Version are you using, I just tryied the current master and cannot reproduce. 
The recipe registration in one file is for my personal feeling a nice approach. It reduces the code because most items are handled equal.

**Comment by FloppyDingo** [comment]
Sorry. 1.11.2h

**Comment by Syndaryl** [comment]
The git repo does not represent the current code base; it has not been updated in quite some time. I believe Pam is still getting acquainted with Git. It very much doesn't represent 1.11.2h - you'd be better off decompiling the JAR file.

**Comment by Peter8Dev** [comment]
I downloaded the 1.11.2h JAR and still cannot "uncook" popcorn or baked sweet potato. (with forge 13.20.0.2294)
bakware + corn = popcorn   and   bakware + sweet potato = baked sweet potato
but
bakware + popcorn = <nothing>   and    bakware + baked sweet potato = <nothing>

Anything else you have to do before or any other mods that may influence each other?

**Comment by FloppyDingo** [comment]
Just proof that this is happening:

http://i.imgur.com/ud1ATCV.png

http://i.imgur.com/ZgLRqbL.png

As for our mod list:

http://i.imgur.com/BD5XHVf.png

(easier to show than to list, hope this is okay)

**Comment by Shiny-andKittens** [comment]
That may be the problem, you have a ton of mods, and some affect food. Try uninstalling all other mods and go onto a new world, then try to use the recipes.

**Comment by FloppyDingo** [comment]
@Shiny-andKittens Okay so this is going to sound crazy:

I spent some time tonight attempting to isolate the bug. I'd remove a handful of mods, then check; and repeat if the bug was still there. Slowly the mods started to dwindle and eventually it was just JEI and Harvestcraft. Surely, JEI couldn't be causing it... and no it wasn't.

...so I load up just harvestcraft... and it's still there.

So I'm baffled. I don't know what to think. So I isolate the harvestcraft jar and download a new one. Same version. Same website (curseforge). Nothing should be different. New jar?

Recipe works just fine.

So somehow, I have an odd jar file of harvestcraft. I've uploaded it to dropbox here if you want to check it out and see wtf happened out of morbid curiosity - I personally am baffled, unless they uploaded a broken version and then switched the versions really quick and I just happened to slip in and grab it before the switch. I have no idea.

https://www.dropbox.com/s/j9htfesb0rq0rc7/Pam%27s%20HarvestCraft%201.11.2h.jar?dl=1
